web/client/src/library/pages/docs/Docs.tsx

```diff
@@ -40,15 +40,21 @@ export default function PageDocs(): JSX.Element {
 
   const list = Array.from(new Set(models.values()))
 
+  const activeItemIndex = useMemo((): number => {
+    return list.findIndex(listItem => {
+      return `/docs/models/${listItem.name}` === pathname
+    })
+  }, [pathname, list])
+
   return (
     <Page
       sidebar={
         <SourceList
-          key={location.pathname}
           by="displayName"
           byName="displayName"
           to={EnumRoutes.IdeDocsModels}
           items={list}
+          activeItemIndex={activeItemIndex}
```

Comment:
I guess it'll be better if we pass  `key/id` (like `active=${EnumRoutes.Docs}/${item.id}` in this case and same for other pages) instead of `index` so we can handle getting actual `index` inside component itself.